lib/surface/compiler/eex_engine.ex

```diff
@@ -0,0 +1,649 @@
+  end
+
+  defp to_token_sequence(nodes) do
+    nodes
+    |> to_dynamic_nested_html()
+    |> List.flatten()
+    |> combine_static_portions()
+  end
+
+  defp generate_buffer([], buffer, state) do
+    state.engine.handle_body(buffer)
+  end
+
+  defp generate_buffer([{:text, chars} | tail], buffer, state) do
+    buffer = state.engine.handle_text(buffer, chars)
+    generate_buffer(tail, buffer, state)
+  end
+
+  defp generate_buffer([expr | tail], buffer, state) do
+    buffer = state.engine.handle_expr(buffer, "=", to_expression(expr, buffer, state))
+    generate_buffer(tail, buffer, state)
+  end
+
+  defp to_expression(nodes, buffer, state)
+
+  defp to_expression([node], buffer, state), do: to_expression(node, buffer, state)
+
+  defp to_expression(nodes, buffer, state) when is_list(nodes) do
+    children =
+      for node <- nodes do
+        to_expression(node, buffer, state)
+      end
+
+    {:__block__, [], children}
+  end
+
+  defp to_expression({:text, value}, _buffer, _state), do: {:safe, value}
+
+  defp to_expression(%AST.AttributeExpr{value: expr}, _buffer, _state), do: expr
+
+  defp to_expression(%AST.Interpolation{value: expr}, _buffer, _state), do: expr
+
+  defp to_expression(%AST.Container{children: children} = container, buffer, state),
+    do: to_expression(children, buffer, state) |> maybe_print_expression(container)
+
+  defp to_expression(
+         %AST.Comprehension{generator: %AST.AttributeExpr{value: generator}, children: children} =
+           comprehension,
+         buffer,
+         state
+       ) do
+    buffer =
+      handle_nested_block(children, buffer, %{
+        state
+        | depth: state.depth + 1,
+          context: [:for | state.context]
+      })
+
+    generator_expr = generator ++ [[do: buffer]]
+
+    {:for, [generated: true], generator_expr}
+    |> maybe_print_expression(comprehension)
+  end
+
+  defp to_expression(
+         %AST.Conditional{condition: %AST.AttributeExpr{value: condition}, children: children} =
+           conditional,
+         buffer,
+         state
+       ) do
+    buffer =
+      handle_nested_block(children, buffer, %{
+        state
+        | depth: state.depth + 1,
+          context: [:if | state.context]
+      })
+
+    {:if, [generated: true], [condition, [do: buffer]]}
+    |> maybe_print_expression(conditional)
+  end
+
+  defp to_expression(
+         %AST.Slot{
+           name: name,
+           props: %AST.Directive{value: %AST.AttributeExpr{value: props_expr}},
+           default: default
+         },
+         buffer,
+         state
+       ) do
+    slot_content_expr =
+      if name == :default do
+        quote generated: true do
+          unquote(at_ref(:inner_content)).(unquote(props_expr))
+        end
+      else
+        slot_name_expr = at_ref(name)
+
+        quote generated: true do
+          for slot <- unquote(slot_name_expr) do
+            slot.inner_content.(unquote(props_expr))
+          end
```

Comment:
```suggestion
          # TODO: For now, we only handle the first since rendering multiple items requires using `:for` directly in the template. 
          # Review this after we adding option `join`.
          Enum.at(unquote(slot_name_expr), 0).inner_content.(unquote(props_expr))
```

Reply:
This was already the current behaviour. We can change it later after we have all tests passing. This fixes 3 of the remaining 6 errors.